packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -731,49 +731,75 @@ class RenderParagraph extends RenderBox
     return _textPainter.size;
   }
 
-  // The offsets for each span that requires custom semantics.
-  final List<int> _inlineSemanticsOffsets = <int>[];
-  // Holds either [GestureRecognizer] or null (for placeholders) to generate
-  // proper semnatics configurations.
-  final List<dynamic> _inlineSemanticsElements = <dynamic>[];
+  /// Collected during [describeSemanticsConfiguration], used by
+  /// [assembleSemanticsNode] and [_combineSemanticsInfo].
+  List<InlineSpanSemanticsInformation> _semanticsInfo;
+
+  /// Combines _semanticsInfo entries where permissible, determined by
+  /// [InlineSpanSemanticsInformation.requiresOwnNode].
+  Iterable<InlineSpanSemanticsInformation> _combineSemanticsInfo() sync* {
```

Comment:
don't use sync* in non-debug code

Reply:
I know you've mentioned this before, but we already have many examples of using this in the framework.  

Is it really better to fully materialize this list and then iterate over it again later?

Reply:
(I'm not trying to do anything that requires memoization here)

Reply:
Refactored to use a list instead.